src/main/java/org/osc/manager/rest/server/api/DeviceApis.java

```diff
@@ -0,0 +1,264 @@
+     *
+     * @return - deleted Device
+     * @throws Exception
+     */
+    @Path("/{deviceId}")
+    @DELETE
+    public void deleteDevice(@PathParam("deviceId") Long deviceId) throws Exception {
+
+        LOG.info("Deleting Device Entity ID...:" + deviceId);
+        this.txControl.required(new Callable<Void>() {
+
+            @Override
+            public Void call() throws Exception {
+
+                DeviceEntity result = DeviceApis.this.em.find(DeviceEntity.class, deviceId);
+                if (result == null) {
+                    LOG.info("Attempt to delete device id:...." + deviceId);
+                    LOG.info("..Device not found, no-op.");
+                    return null;
+                }
+
+                DeviceApis.this.em.remove(result);
+                return null;
+            }
+        });
+    }
+
+    /**
+     * Lists the Device Id's
+     *
+     * @return - Device Id's
+     * @throws Exception
+     */
+    @GET
+    public List<String> getDeviceIds() throws Exception {
+
+        LOG.info("Listing Device Ids'");
+
+        List<? extends ManagerDeviceElement> devices = this.api.listDevices();
+        List<String> deviceList = new ArrayList<String>();
+        if (devices.isEmpty()) {
+            return deviceList;
+        }
+        deviceList = devices.stream().map(ManagerDeviceElement::getId).collect(Collectors.toList());
+        return deviceList;
+    }
+
+    /**
+     * Gets the Device for a given device Id
+     *
+     * @return - Device
+     *
+     */
+    @Path("/{deviceId}")
+    @GET
+    public DeviceEntity getDevice(@PathParam("deviceId") Long deviceId) throws Exception {
+
+        LOG.info("getting Device for ID...:" + deviceId);
+        return (DeviceEntity) this.api.getDeviceById(Long.toString(deviceId));
+    }
+
+    /**
+     * Creates the Device Member for a given Device
+     *
+     * @return device member id
+     * @throws Exception
+     */
+    @Path("/{deviceId}/members")
+    @POST
+    public String createDeviceMember(@PathParam("deviceId") Long deviceId, DeviceMemberEntity entity) throws Exception {
+
+        LOG.info("Creating Device Members...:" + entity.getName());
+        DeviceEntity device = new DeviceEntity();
+        entity.setParent(device);
+        device.setId(deviceId);
+        return this.api.createDeviceMember(entity);
+    }
+
+    /**
+     * Updates the Device Member for a given Device Id and the Device Member Id
+     *
+     * @return - updated Device Member
+     * @throws Exception
+     */
+    @Path("/{deviceId}/members/{memberId}")
+    @PUT
+    public DeviceMemberEntity updateDeviceMember(@PathParam("deviceId") Long deviceId,
+            @PathParam("memberId") Long memberId, DeviceMemberEntity entity) throws Exception {
+
+        LOG.info("Updating for Device Member with Device Member Id...:" + memberId);
+        LOG.info("and Device Id...:" + deviceId);
+        entity.setId(memberId);
+        this.api.updateDeviceMember(entity);
```

Comment:
ensure to pass the `deviceId` here to enforce the parent relationship, see other comments.